crates/bevy_ecs/src/query/filter.rs

```diff
@@ -757,21 +763,274 @@ unsafe impl<T: Component> QueryFilter for Added<T> {
     }
 }
 
-/// A filter on a component that only retains results the first time after they have been added or mutably dereferenced.
+/// A [`QueryFilter`] on a component that only retains results after the most recent time the component was mutably dereferenced.
+/// A common use for this filter is avoiding redundant work when values have not changed.
+///
+/// # Notes about usage
+///
+/// - Simply *mutably dereferencing* a component is considered a change ([`DerefMut`](std::ops::DerefMut)).
+///   Bevy does not compare components to their previous values.
+/// - This filter includes changes that happened before the first time this `Query` was run.
+/// - To retain all results without filtering but still check whether they were mutated after the
+///   system last ran, use [`Ref<T>::is_mutated`](crate::change_detection::Ref) instead.
+///
+/// # Comparison with [`Added<T>`] and [`Changed<T>`]
+///
+/// `Mutated<T>` is effectively implemented as `(Changed<T>, Not<Added<T>>)`,
+/// which means it only reacts to changes that aren't [additions](Added) (i.e. only mutable dereferences).
```

Comment:
We need to be clearer in the docs here: this is Mutated && !Added, not Mutated (regardless of whether or not the component has been added). This means that even if a mutation occurred, if an addition *also* occurred, this will skip your entity.